### PR TITLE
fix: trap SQLAlchemy common exceptions & throw 422 error instead

### PR DESCRIPTION
**Pull request by diegomedina248**
### SUMMARY
This PR adds exception handling for the most common SQLAlchemy exceptions, returning a 422 response instead of a plain 500 error.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
_not applicable_

### TESTING INSTRUCTIONS
This can only be verified in code, raising one of the added errors and inspecting the response, ensuring it's a 422 one.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/base.py [hunk 1]
```diff
@@ -231,6 +231,9 @@ def wraps(self: "BaseSupersetView", *args: Any, **kwargs: Any) -> FlaskResponse:
             return json_error_response(
                 utils.error_msg_from_exception(ex), status=cast(int, ex.code)
             )
+        except (exc.IntegrityError, exc.DatabaseError, exc.DataError) as ex:
+            logger.exception(ex)
+            return json_error_response(utils.error_msg_from_exception(ex), status=422)
```
Comment (codemaster08240328) [inline superset/views/base.py:236]: Returning `422` status code for `DatabaseError` and `IntegrityError` is not good. Because these errors are not related to the entity which is sent from Client side. 

`500` is more acceptable for these errors. 
`422` is good for `DataError`

Check here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
https://docs.sqlalchemy.org/en/14/errors.html

**Review by codemaster08240328 — CHANGES_REQUESTED** [review]

**Review by rusackas — APPROVED** [review]
